### ISSUE TITLE
Filepicker Web - how to get path?

### ISSUE THREAD

**Issue by guit4eva** [miguelpruivo/flutter_file_picker#361]
When using file_picker_web.dart, the image is returned as [Object File], for example:

> print(images)
> // Android: [File: '/data/user/0...../example.png', File: '/data/user/0...../example2.png']
> // Web: [[object File], [object File]]

How can I get the path of the image when using web so that I can display it in the app? 

**note: img.relativePath returns an empty string**

```
import 'package:file_picker_web/file_picker_web.dart' as webFilePicker;

 if (kIsWeb) {
        await webFilePicker.FilePicker.getMultiFile(type: FileType.image).then(
          (chosenImages) async {
            if (chosenImages != null)
              for (var img in chosenImages) {
                print(img.runtimeType);
                images.add(img);
              }
          },
        );
      } else if (Platform.isAndroid) {
        await FilePicker.getMultiFile(type: FileType.image).then(
          (chosenImages) async {
            if (chosenImages != null)
              for (var img in chosenImages) {
                print(img.runtimeType);
                images.add(img);
              }
          },
        );
      } 
```

**Comment by miguelpruivo** [comment]
@guit4eva I think that you need to get the bytes from it as web doesn't actually use relative paths for it. I'm sorry, but I'm not really into it as I don't actually work with Flutter Web (only for plugin maintenance). This [answer](https://stackoverflow.com/a/57802163/6919963) might help you.

**Comment by miguelpruivo** [comment]
@guit4eva let me know if it worked for you.

**Comment by Wylarel** [comment]
Hey, I have the same issue, but I need a IO File like when I use `file_picker` on Android so I can upload it to Firebase Storage. My files are not only images, they can be any format. How should I convert HTML File (I think that's the output I get from `FilePicker.getFile()` when using `file_picker_web`) to IO File ? Cast doesn't seem to work, all I know is that when I use a cast I get an error (in the web js console):
```dart
Uncaught TypeError: Instance of 'minified:dB': type 'minified:dB' is not a subtype of type 'minified:axV'
    at Object.c (js_helper.dart:1129)
    at Object.akq (rti.dart:1016)
    at h9.au2 [as a] (rti.dart:1009)
    at h9.aud (rti.dart:933)
    at upload_download_view.dart:88
    at ac0.a (async_patch.dart:315)
    at ac0.$2 (main.dart.js:27690)
    at abo.$1 (async_patch.dart:291)
    at a9W.a4W (main.dart.js:28463)
    at a9W.Aa (zone.dart:1451)
c @ js_helper.dart:1129
$0 @ main.dart.js:28434
auB @ schedule_microtask.dart:41
auO @ schedule_microtask.dart:50
$1 @ async_patch.dart:48
aw5 @ js_helper.dart:1825
(anonymous) @ js_helper.dart:1857
childList (async)
$1 @ async_patch.dart:62
akR @ async_patch.dart:27
of @ zone.dart:1257
QF @ main.dart.js:27976
ib @ future_impl.dart:540
cg @ future_impl.dart:43
$1 @ file_picker_web.dart:39
$1 @ html_dart2js.dart:37188
aw5 @ js_helper.dart:1825
(anonymous) @ js_helper.dart:1857
```
But I guess those error logs are not useful

**Comment by miguelpruivo** [comment]
@Wylarel once again, you need to use conditional imports and either use `html.File` or `io.File`. Mixing both won’t work.

**Comment by Wylarel** [comment]
> you need to use conditional imports
That's what I do!

The thing is that Firebase Storage **requires** io.File, it doesn't work when I try to upload a html.File. So I need to convert the html.File from FilePicker.getFile() to a io.File but a simple cast doesn't work.

**Comment by miguelpruivo** [comment]
@Wylarel the problem is that firebase storage doesn't yet support web, that's why you _can't_ use it. They'll need to support uploads with byte data for web. There's no way a `io.File` instance can work in web environment — this is, transpiled to dart.js.

**Comment by Wylarel** [comment]
Right, if anyone is wondering, using storage `firebase/firebase.dart` instead of `firebase_storage/firebase_storage.dart` worked, there I can send a html.File. ([How to use it](https://dev.to/happyharis/flutter-web-firebase-storage-2ac1))

For the original issue, yeah it seems impossible to get the absolute or relative path, you can only get the name from html.File...

**Comment by guit4eva** [comment]
> @guit4eva let me know if it worked for you.

Thanks for that. I managed to get the images to display, but unfortunately adding them to a list with byte data is almost impossible due to the size (starts freezing after adding a few), and doesn't look like uploading to firebase storage is possible after reading through the other comments here. So close, but no cigar! Will have to wait for better support. I'll post back here if I manage to find a solution in the meantime

**Comment by miguelpruivo** [comment]
I guess we're coming across a Flutter Web bottleneck then. Let me know if I could make it easier to you.
